components/support/locale/src/main/java/mozilla/components/support/locale/LocaleManager.kt

```diff
@@ -77,6 +77,18 @@ object LocaleManager {
         return ConfigurationCompat.getLocales(config).get(0)
     }
 
+    /**
+     * Update the locale for the configuration of the app context's resources
+     */
+    @Suppress("DEPRECATION")
```

Comment:
I'm leaving this here to initiate a bit of talk, in case somebody comes up with a better solution. The `updateConfiguration` method got deprecated, however I currently don't see a decent solution for how to update the appContext's configuration (injecting localized strings in singletons from places with activity context doesn't seem clean, and using the activity context in the singleton is a full-blown leak). A usecase for why it's needed is described in the #5631  task